### ISSUE TITLE
publish release

### ISSUE THREAD

**Issue by a-detiste** [rtissera/libchdr#101]
Hi,

Could you publish a release ?
I'd like to package this for Debian and strip the vendored copies from `ares`, `mrboom` and `retroarch`.

Having versionned releases makes it much easier.

Greetings,

**Comment by rtissera** [comment]
Hello Alexandre,

Yes, for sure, that would be great indeed.
I need to add the recent DVD support introduced in MAME 0.254 or 0.255,
can't remember, and we could go for a versioned release.

Beware that some vendored copies might have subtle changes, especially in
the libretro area.

Best Regards,
Romain Tisserand

**Comment by Neustradamus** [comment]
Any news on it?

**Comment by dmstocking** [comment]
@rtissera Is there a specification for the DVD support with CHD? I wouldn't mind implementing it, but I can't really find anything besides the mame source code. I could use it. It is just a little annoying to figure it out from the code.

**Comment by rtissera** [comment]
DVD support has been added a while ago.

**Comment by dmstocking** [comment]
@rtissera Oh. Apologies. So https://github.com/libretro/RetroArch/issues/16145 seems to believe that libchdr needed to add support. That was said about a year ago. I didn't see anything in the commits that seemed like it was implemented within the last year. Is this just incorrect on RetroArch's side? Do they just need to update libchdr? I could go do that for them. 😄

**Comment by nickkeane** [comment]
Hashes are currently not properly generated for PPSSPP games using the CHD format. Presumably libchdr has not been updated to support this feature and/or RetroArch and the PPSSPP core have not incorporated an updated version of libchdr for this yet.

RetroArch: 1.19.1
PPSSPP Core: 63c7c73
OS: Windows 11 x64
CPU: Intel i9 14900H
GPU: nVidia RTX 4070 Laptop

<img width="949" alt="image" src="https://github.com/user-attachments/assets/fd6c0c05-99d4-4160-827b-354182635e48" />

See also https://github.com/libretro/RetroArch/issues/16145#issuecomment-2573653774

**Comment by rtissera** [comment]
RetroArch and libretro-common use a very old, forked libchdr codebase...

**Comment by orbea** [comment]
@rtissera Now that [Geolith supports libchdr](https://gitlab.com/jgemu/geolith/-/blob/master/Makefile?ref_type=heads#L65) I would like to support this in the Gentoo ebuild as well, but its difficult unless libchdr maintains git tags for releases, can you please do that?

**Comment by rtissera** [comment]
@orbea yes it's about time.... am on it right now, fixing potential security issues and so on.

**Comment by rtissera** [comment]
@orbea @a-detiste @Neustradamus 
Released v0.3.0: https://github.com/rtissera/libchdr/releases/tag/v0.3.0

Tag on `93d8c23`. `SOVERSION = 0` preserved, so the installed SONAME stays `libchdr.so.0` — no Debian package rename needed.

cc @a-detiste: the armhf time64 `ftello64` patch (#117) and the strict-C11 `-Werror=implicit-function-declaration` failure (#92) are both fixed upstream in this tag, along with `CHD_MAX_FILE_SIZE` bumped to 1 TB (#147) and three CHD-parser security fixes (#148). You can drop the out-of-tree patches when rebuilding against v0.3.0.

**Comment by orbea** [comment]
@rtissera Thank you very much! Getting stuff merged into Gentoo can be a slow process at times, but I will work on it when the current PR with the Geolith update is done. https://github.com/gentoo/gentoo/pull/46120

**Comment by Neustradamus** [comment]
@rtissera: Thanks for this first version!

**Comment by orbea** [comment]
I created a PR to add libchdr back to Gentoo. https://github.com/gentoo/gentoo/pull/46190

**Comment by orbea** [comment]
Gentoo now has llibchdr again! https://github.com/gentoo/gentoo/commit/583345bf92f12ff50a3803e59886e44e8b5bf285

I also made a keywording request where arm, arm64, pcc, ppc64 and x86 will hopefully be tested to match the rest of the associated JGEMU packages. https://bugs.gentoo.org/976838

**Comment by Neustradamus** [comment]
@orbea: Good job, thanks!
